internal/app/pactproxy/interaction.go

```diff
@@ -352,22 +356,16 @@ func (i *Interaction) EvaluateConstraints(request requestDocument, interactions
 			}
 		}
 
-		actual := ""
 		val, err := jsonpath.Get(request.encodeValues(constraint.Path), map[string]interface{}(request))
 		if err != nil {
-			log.Warn(err)
-		}
-		if reflect.TypeOf(val) == reflect.TypeOf([]interface{}{}) {
-			log.Infof("skipping matching on []interface{} type for path '%s'", constraint.Path)
+			violations = append(violations,
+				fmt.Sprintf("constraint path %q cannot be resolved within request: %q", constraint.Path, err))
+			result = false
 			continue
 		}
-		if err == nil {
-			actual = fmt.Sprintf("%v", val)
-		}
 
-		expected := fmt.Sprintf(constraint.Format, values...)
-		if actual != expected {
-			violations = append(violations, fmt.Sprintf("value '%s' at path '%s' does not match constraint '%s'", actual, constraint.Path, expected))
+		if err := constraint.Check(values, val); err != nil {
```

Comment:
Nit: I slightly prefer original variable naming, tbh (i.e. `actual` vs. `expected`).

Reply:
`expected` & `actual` got moved inside `Check`. So `values` & `val` here are the raw data that was previously used to create those. They are very confusingly named though, and now `expected` & `actual` are freed up they work well here :+1: